src/Action_Hbond.cpp

```diff
@@ -76,6 +80,9 @@ Action::RetType Action_Hbond::Init(ArgList& actionArgs, ActionInit& init, int de
   useAtomNum_ = actionArgs.hasKey("printatomnum");
   acut_ = actionArgs.getKeyDouble("angle",135.0);
   noIntramol_ = actionArgs.hasKey("nointramol");
+  if (actionArgs.hasKey("atomindex")) {
+    AHDIndices_ = (DataSet_integer*) init.DSL().AddSet(DataSet::INTEGER, "atomindex");
```

Comment:
@drroe 

I change to yours and got this

`Internal Error: DataSet has no name.
Error setting up data set [atomindex].
`